### PR TITLE
Fix for using 'on_behalf_of' method in multi-threaded environment

### PR DESCRIPTION
**Pull request by rjnienaber**
This should address issue #5. The session for a thread is now saved in thread local storage. This should prevent multiple threads from clobbering the on_behalf_of value. Before we go to v1.0, we should consider rewriting this in a more OOP style.

### CODE REVIEW

File: currencycloud/__init__.py [hunk 1]
```diff
@@ -14,55 +15,61 @@
...
-    __session = None
+    __ensure_thread_session()
+    thread_storage.session = None
     token = None
 
 
 @contextmanager
 def on_behalf_of(contact_id):
-    global __session
+    __ensure_thread_session
```
Comment (danielsamuels) [inline currencycloud/__init__.py]: Should this be `__ensure_thread_session()`?

Comment (rjnienaber) [inline currencycloud/__init__.py]: :grimacing: I did miss that but after wondering why it was still working without that call, I realised that the `session` method would set it up. I've removed it in a subsequent commit.

**Comment by teodorgarzdin** [comment]
Is this going to be merged in anytime soon?

**Comment by tpyo** [comment]
@rjnienaber what's the status on this?

**Comment by rjnienaber** [comment]
@tpyo, @teodorgarzdin, I no longer have merge access so I'm deferring this to @liamm.

**Pull request closed without merging**